ngraph/frontend/paddlepaddle/src/frontend.cpp

```diff
@@ -157,41 +160,108 @@ namespace ngraph
             return std::make_shared<ngraph::Function>(result_nodes, parameter_nodes);
         }
 
-        InputModel::Ptr FrontEndPDPD::load_from_file(const std::string& path) const
-        {
-            return load_from_files({path});
-        }
-
-        InputModel::Ptr FrontEndPDPD::load_from_files(const std::vector<std::string>& paths) const
+        bool FrontEndPDPD::supported_by_variants(
+            const std::vector<std::shared_ptr<Variant>>& variants) const
         {
-            if (paths.size() == 1)
+            // Validating first path, it must contain a model
+            PDPD_ASSERT(variants.size() > 0, "Variants should contain at least one path/stream");
+            if (is_type<VariantWrapper<std::string>>(variants[0]))
             {
-                // The case when folder with __model__ and weight files is provided or .pdmodel file
-                return std::make_shared<InputModelPDPD>(paths[0]);
+                std::string suffix = ".pdmodel";
+                std::string model_path =
+                    as_type_ptr<VariantWrapper<std::string>>(variants[0])->get();
+                if (model_path.length() < suffix.length() ||
+                    (0 != model_path.compare(
+                              model_path.length() - suffix.length(), suffix.length(), suffix)))
+                {
+                    model_path += "/__model__";
+                }
+                std::ifstream model_str(model_path, std::ios::in | std::ifstream::binary);
+                // It is possible to validate here that protobuf can read model from the stream,
+                // but it will complicate the check, while it should be as quick as possible
+                return model_str && model_str.is_open();
             }
-            else if (paths.size() == 2)
+            else if (is_type<VariantWrapper<std::shared_ptr<std::istream>>>(variants[0]))
             {
-                // The case when .pdmodel and .pdparams files are provided
-                std::ifstream model_stream(paths[0], std::ios::in | std::ifstream::binary);
-                FRONT_END_INITIALIZATION_CHECK(model_stream && model_stream.is_open(),
-                                               "Cannot open model file.");
-                std::ifstream weights_stream(paths[1], std::ios::in | std::ifstream::binary);
-                FRONT_END_INITIALIZATION_CHECK(weights_stream && weights_stream.is_open(),
-                                               "Cannot open weights file.");
-                return load_from_streams({&model_stream, &weights_stream});
+                // Validating first stream, it must contain a model
+                std::shared_ptr<std::istream> p_model_stream =
+                    as_type_ptr<VariantWrapper<std::shared_ptr<std::istream>>>(variants[0])->get();
+                paddle::framework::proto::ProgramDesc fw;
+                return fw.ParseFromIstream(p_model_stream.get());
             }
```

Comment:
shall it return 'false' in the end?

Reply:
fw.ParseFromIstream will return false if it cant parse